nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2584,8 +2584,8 @@ pub(crate) mod test {
 
         assert_eq!(summary.sleds_added.len(), 0);
         assert_eq!(summary.sleds_removed.len(), 0);
-        assert_eq!(summary.sleds_modified.len(), 4);
-        assert_eq!(summary.sleds_unchanged.len(), 1);
+        assert_eq!(summary.sleds_modified.len(), 3);
+        assert_eq!(summary.sleds_unchanged.len(), 2);
```

Comment:
I believe this test was "wrong" (in that a sled was showing up as modified only because we were dropping its datasets): The sled that moved from `modified` to `unchanged` is `sled 68d24ac5-f341-49ea-a92a-0381b52ab387` in `nexus/reconfigurator/planning/tests/output/planner_nonprovisionable_1_2.txt`. The changes in the diff prior to this PR are exclusively "all the datasets were removed"; now that that isn't happening, the sled is unchanged.